src/main/java/com/matt/forgehax/mods/DerpMod.java

```diff
@@ -0,0 +1,244 @@
+          .min(0f)
+          .max(1f)
+          .defaultTo(.25f)
+          .build();
+
+  public final Setting<Float> hitPan =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("hit_pan")
+          .description("Chance for any hit to be the off hand")
+          .min(0f)
+          .max(1f)
+          .defaultTo(.5f)
+          .build();
+
+  private void sendRotatePacket(EntityPlayer player, float pitch, float yaw) {
+    ignoreAndSend(new CPacketPlayer.Rotation(yaw, pitch, player.onGround));
+  }
+
+  private void rotateDisabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player == null) return;
+
+    sendRotatePacket(player, player.rotationPitch, player.rotationYawHead);
+  }
+
+  private CPacketEntityAction.Action actionFromSneaking(boolean sneaking) {
+    return sneaking
+        ? CPacketEntityAction.Action.START_SNEAKING
+        : CPacketEntityAction.Action.STOP_SNEAKING;
+  }
+
+  private void sendSneakPacket(EntityPlayer player, boolean sneaking) {
+    this.sneaking = sneaking;
+    ignoreAndSend(new CPacketEntityAction(player, actionFromSneaking(sneaking)));
+  }
+
+  private void sneakDisabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player == null) return;
+
+    sendSneakPacket(player, player.isSneaking());
+  }
+
+  @Override
+  protected void onDisabled() {
+    if (rotate.get()) rotateDisabled();
+    if (sneak.get()) sneakDisabled();
+  }
+
+  private void sendHitPacket(EnumHand hand) {
+    ignoreAndSend(new CPacketAnimation(hand));
+  }
+
+  private boolean chance(float chance) {
+    return chance == 1f || (chance != 0f && Math.random() <= chance);
+  }
+
+  @SubscribeEvent
+  public void onLocalPlayerUpdate(LocalPlayerUpdateEvent event) {
+    EntityPlayer player = (EntityPlayer) event.getEntityLiving();
+    if (player == null) return;
+    Random rng = player.getRNG();
+
+    float effectiveSpeed = speed.get();
+
+    if (timerSync.get() && timerMod != null && timerMod.isEnabled())
+      effectiveSpeed /= timerMod.factor.get();
+
+    error += effectiveSpeed;
+    int iter = (int) error;
+    error -= iter;
+
+    for (int i = 0; i < iter; i++) {
+      if (rotate.get() && chance(rotateChance.get()))
+        sendRotatePacket(player, rng.nextInt(180) - 90, rng.nextInt(360) - 180);
+
+      if (sneak.get() && chance(sneakChance.get()))
+        sendSneakPacket(player, !sneaking);
+
+      if (hit.get() && chance(hitChance.get()))
+        sendHitPacket(chance(hitPan.get()) ? EnumHand.OFF_HAND : EnumHand.MAIN_HAND);
+    }
+  }
+
+  @SubscribeEvent
+  public void onPacketSending(PacketEvent.Outgoing.Pre event) {
+    Packet<?> packet = event.getPacket();
+    if (isIgnored(packet)) return;
+
+    if (rotate.get() && (packet instanceof CPacketPlayer.Rotation || packet instanceof CPacketPlayer.PositionRotation)) {
+      event.setCanceled(true);
+
+      if (packet instanceof CPacketPlayer.PositionRotation) {
```

Comment:
What is this for? Either way, it probably won't be needed once you use the position rotation service.

Reply:
This is to block rotation packets from being sent to the server so Derp can inject its own, and only its own.

Reply:
This injects Position packets in place of PositionRotation ones. I'll replace ignoreAndSend with a regular send as this should really not be ignored.